packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,443 @@
+      immediate
+    >
+      <va-input
+        v-model="input"
+        label="immediate validation"
+        :rules="[value => value === 'hello' || 'should be hello']"
+      />
+    </va-form>
+    [deffault]
+    <va-form
+      class="flex max-w-sm"
+    >
+      <va-input
+        v-model="input"
+        label="validation on change"
+        :rules="[value => value === 'hello' || 'should be hello']"
+      />
+    </va-form>
+  `,
+})
+
+ValidationRules.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'immediate validation' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'validation on change' }) as HTMLElement
+
+  await step('Displays error message', async () => {
+    expect(firstInput.getAttribute('aria-invalid')).toEqual('true')
+  })
+
+  await step('Does not display error message', async () => {
+    expect(secondInput.getAttribute('aria-invalid')).toEqual('false')
+  })
+}
+
+export const HideErrorMessages = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'error' }),
+  template: `
+    [hidden]
+    <va-form
+      class="flex max-w-sm"
+      ref="rulesFormRef"
+      immediate
+      :hideErrorMessages="true"
+    >
+      <va-input
+        v-model="input"
+        label="with no error message"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+    [default]
+    <va-form
+      class="flex max-w-sm"
+      ref="rulesFormRef"
+      immediate
+      :hideErrorMessages="false"
+    >
+      <va-input
+        v-model="input"
+        label="with error message"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+  `,
+})
+
+HideErrorMessages.play = async ({ step }) => {
+  const [firstDefinedInput, secondDefineInput] = document.querySelectorAll('div.va-input-wrapper > div.va-message-list')
+
+  await step('Does not display validation error', async () => {
+    expect(firstDefinedInput).toBeDefined()
+  })
+
+  await step('Displays validation error', async () => {
+    expect(secondDefineInput).toBeUndefined()
+  })
+}
+
+export const HideErrors = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'error' }),
+  template: `
+    [hidden]
+    <va-form
+      class="flex max-w-sm"
+      immediate
+      :hideErrors="true"
+    >
+      <va-input
+        v-model="input"
+        label="with no error highlight"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+    [default]
+    <va-form
+      class="flex max-w-sm"
+      immediate
+      :hideErrors="false"
+    >
+      <va-input
+        v-model="input"
+        label="with error highlight"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+  `,
+})
+
+HideErrors.play = async ({ step }) => {
+  const [firstInput, secondInput] = document.querySelectorAll('div.va-input-wrapper')
+
+  await step('Does not highlight invalid input', async () => {
```

Comment:
If we don't have a role for errored input - that could be considered input's bug.